etrago/tools/constraints.py

```diff
@@ -1183,6 +1194,211 @@ def _capacity_factor_per_gen_cntr_nmp(self, network, snapshots):
                 )
 
 
+def read_max_gas_generation(self):
+    """Return the values limiting the gas production in Germany
+
+    Read max_gas_generation_overtheyear from
+    scenario.egon_scenario_parameters if the table is available in the
+    database and return the dictionnary containing the values needed
+    for the constraints to limit the gas production in Germany,
+    depending of the scenario.
+
+    Returns
+    -------
+    arg: dict
+
+    """
+    scn_name = self.args["scn_name"]
+    arg_def = {
+        "eGon2035": {"CH4": 36000000, "biogas": 10000000},
+        "eGon100RE": {"biogas": 14450103},
+    }
```

Comment:
It would be nice to have some information about these values, e.g. the unit and a source.

Reply:
Thanks for the remark!
Changed in https://github.com/openego/eTraGo/pull/539/commits/63c476f2ae7602c37b63c4613acc9728682ecd6b.